pkg/sql/tenant.go

```diff
@@ -185,8 +185,35 @@ func CreateTenantRecord(
 	)
 }
 
-// GetTenantRecord retrieves a tenant in system.tenants.
-func GetTenantRecord(
+// GetTenantRecordByName retrieves a tenant with the provided name from
+// system.tenants.
+func GetTenantRecordByName(
+	ctx context.Context, execCfg *ExecutorConfig, txn *kv.Txn, tenantName roachpb.TenantName,
+) (*descpb.TenantInfo, error) {
+	if !execCfg.Settings.Version.IsActive(ctx, clusterversion.V23_1TenantNames) {
+		return nil, errors.Newf("tenant names not supported until upgrade to %s or higher is completed",
+			clusterversion.V23_1TenantNames.String())
+	}
+	row, err := execCfg.InternalExecutor.QueryRowEx(
+		ctx, "activate-tenant", txn, sessiondata.NodeUserSessionDataOverride,
+		`SELECT info FROM system.tenants WHERE name = $1`, tenantName,
+	)
+	if err != nil {
+		return nil, err
+	} else if row == nil {
+		return nil, pgerror.Newf(pgcode.UndefinedObject, "tenant \"%q\" does not exist", tenantName)
```

Comment:
[nit] I think this will end up double-quoting the tenant name since %q also produces quotes.

Reply:
nice catch, fixed